src/Psl/DateTime/Duration.php

```diff
@@ -0,0 +1,729 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\DateTime;
+
+use JsonSerializable;
+use Psl\Comparison;
+use Psl\Math;
+use Psl\Str;
+use Stringable;
+
+/**
+ * Defines a representation of a time duration with specific hours, minutes, seconds,
+ * and nanoseconds.
+ *
+ * All instances are normalized as follows:
+ *
+ * - all non-zero parts (hours, minutes, seconds, nanoseconds) will have the same sign
+ * - minutes, seconds will be between -59 and 59
+ * - nanoseconds will be between -999999999 and 999999999 (less than 1 second)
+ *
+ * For example, Duration::hours(2, -183) normalizes to "-1 hour(s), -3 minute(s)".
+ *
+ * @implements Comparison\Comparable<Duration>
+ * @implements Comparison\Equable<Duration>
+ *
+ * @immutable
+ */
+final readonly class Duration implements Comparison\Comparable, Comparison\Equable, JsonSerializable, Stringable
+{
+    /**
+     * Initializes a new instance of Duration with specified hours, minutes, seconds, and
+     * nanoseconds.
+     *
+     * @param int $hours
+     * @param int<-59, 59> $minutes
+     * @param int<-59, 59> $seconds
+     * @param int<-999999999, 999999999> $nanoseconds
+     *
+     * @pure
+     */
+    private function __construct(
+        private int $hours,
+        private int $minutes,
+        private int $seconds,
+        private int $nanoseconds
+    ) {
+    }
+
```

Comment:
it could make sense to add a `parse()` method  that takes a format like '+1 hours 30 minutes' that `strtotime()` currently accepts or a format that `DateInterval::createFromDateString('1 day')` accepts. Or maybe also a compatibility layer with `DateInterval`.

Reply:
>'+1 hours 30 minutes' that strtotime() currently accepts or a format that DateInterval::createFromDateString('1 day')

we could 🤔 

>maybe also a compatibility layer with DateInterval

that would be added after this PR as part of the `Interp` component.

Reply:
Just pasting this here:
https://github.com/brick/date-time/blob/324425632ac5933b7253bde65520ca13de850f7a/src/Duration.php#L53-L131

Brick uses this parser for parsing its duration.